metaworld/envs/mujoco/multitask_env.py

```diff
@@ -101,14 +101,18 @@ def _check_env_list(self):
 
     @property
     def observation_space(self):
-        """ TODO not complete, needs support for different observation types"""
+        """TODO not complete, needs support for different observation types."""
         if self._obs_type == "plain":
             obs_dim = self._max_obs_dim
         elif self._obs_type == "with_goal_id":
             obs_dim = self._max_obs_dim + _NUM_METAWORLD_ENVS
         else:
             raise Exception("invalid obs_type, obs_type was {}".format(self._obs_type))
-        return Box(low=-np.inf, high=np.inf, shape=(obs_dim,))
+        observation_space = Box(
+            self._augment_observation(self.active_env.observation_space.low),
```

Comment:
so you need to verify that all of the environments have the same observation_space, or else this can be wrong in very unexpected ways.

i think @avnishn has worked on unifying the observation spaces, but i'm not sure that it's done.

it might be simpler to just statically declare the observation space

Reply:
When you say statically declare observation space, what do you mean? I couldn't leave the old +-inf space because it needs to be a callable for the check. Should I just return self.active_env.observation_space then for now?

Reply:
@ryanjulian just a quick clarification. should I make a change here now (like the one stated above) or just wait until the observation spaces are unified?

Reply:
So at the end of the day, all 50 environments need to share an observation space, because (in theory) a MultitaskEnv is a single environment, and the observation space of an environment cannot change.

I am suggesting we just determined which observation space can contain all 50 environments, write it down ("statically declare") and then check in tests that they all conform to this. It's much simpler than running code every time to determine a dynamic intersection of the observation spaces, which is error-prone and wasteful.

Reply:
Understood. Thanks for the clarification.

Reply:
@ryanjulian I have made changes to this PR so the test works without making changes to the MultiTaskEnv class.